### PR TITLE
COMMERCE-10610 Check channel rel permission

### PR DESCRIPTION
**Pull request by jaydawu**
https://issues.liferay.com/browse/COMMERCE-10610
Resend from https://github.com/liferay-commerce/liferay-portal/pull/3556

### CODE REVIEW

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVisibility.testcase [hunk 1]
```diff
@@ -55,6 +55,83 @@ definition {
...
+				accountType = "Business",
+				agreeToTermsAndAnswerReminderQuery = "true",
+				createNewAccount = "true",
+				requireReset = "false",
+				userEmailAddress = "buyer@liferay.com",
+				userFirstName = "Buyer",
+				userLastName = "User",
+				userRole = "Buyer",
+				userScreenName = "buyeruser",
+				userSiteMembership = "Minium");
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVisibility.testcase:78]: I don't think we need to create an account with a buyer user, just creating a regular user is enough. Thanks!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVisibility.testcase [hunk 2]
```diff
@@ -55,6 +55,83 @@ definition {
 			name = "Channels");
 	}
 
+	@description = "This is a test for COMMERCE-10610. As a buyer user, I can view the default channels with no results using the view permissions for products and catalogs of the admin panel"
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVisibility.testcase:58]: Based on my following comment, please adjust the description accordingly. Thanks!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVisibility.testcase [hunk 3]
```diff
@@ -55,6 +55,83 @@ definition {
...
+	@priority = 3
+	test CanViewDefaultNoChannelResultsWithViewPermissionsForProductsAndCatalogs {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVisibility.testcase:60]:
```suggestion
	test CanAssertEmptyProductChannelsVisibilityWithProductsAndCatalogsViewPermissions {
```

**Review by lorenzolapescara — CHANGES_REQUESTED** [review]
@jaydawu please have a look at my comments, thanks!

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVisibility.testcase [hunk 4]
```diff
@@ -55,6 +55,82 @@ definition {
...
+				accountName = "Commerce Account",
+				accountType = "Business",
+				agreeToTermsAndAnswerReminderQuery = "true",
+				createNewAccount = "true",
+				requireReset = "false",
+				userEmailAddress = "testuser@liferay.com",
+				userFirstName = "Test",
+				userLastName = "User",
+				userScreenName = "testuser",
+				userSiteMembership = "Minium");
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVisibility.testcase:77]:
```suggestion
		JSONUser.addUser(
			userEmailAddress = "testuser@liferay.com",
			userFirstName = "Test",
			userLastName = "User",
			userScreenName = "testuser");

		JSONUser.setFirstPassword(
			agreeToTermsAndAnswerReminderQuery = "true",
			requireReset = "false",
			userEmailAddress = "testuser@liferay.com");
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVisibility.testcase [hunk 5]
```diff
@@ -55,6 +55,82 @@ definition {
...
+				roleTitle = "User");
+		}
+
+		task ("When the user logs in and goes to the Visibility tab of ABS Sensor product in the products admin panel") {
+			User.logoutPG();
+
+			CommerceLogin.miniumLogin(
+				password = "test",
+				urlAppend = "web/minium",
+				userEmailAddress = "testuser@liferay.com");
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVisibility.testcase:118]:
```suggestion
			User.logoutAndLoginPG(
				userLoginEmailAddress = "testuser@liferay.com",
				userLoginFullName = "Test User");
```

**Review by lorenzolapescara — CHANGES_REQUESTED** [review]
@jaydawu please check the new comments, thanks!

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:forward

**Pull request closed without merging**